src/typemap.c

```diff
@@ -1098,9 +1090,9 @@ static void jl_typemap_list_insert_sorted(jl_typemap_entry_t **pml, jl_value_t *
     jl_value_t *pa = parent;
     while (l != (void*)jl_nothing) {
         if (!l->isleafsig) { // quickly ignore all of the leafsig entries (these were handled by caller)
-            if (jl_args_morespecific((jl_value_t*)newrec->sig, (jl_value_t*)l->sig)) {
+            if (jl_type_morespecific((jl_value_t*)newrec->sig, (jl_value_t*)l->sig)) {
                 if (l->simplesig == (void*)jl_nothing ||
-                    newrec->simplesig != (void*)jl_nothing || !sigs_eq((jl_value_t*)l->sig, (jl_value_t*)newrec->sig, 1)) {
+                    newrec->simplesig != (void*)jl_nothing || !jl_types_equal((jl_value_t*)l->sig, (jl_value_t*)newrec->sig)) {
                     // might need to insert multiple entries for a lookup differing only by their simplesig
                     // when simplesig contains a kind
                     // TODO: make this test more correct or figure out a better way to compute this
```

Comment:
in the past, this comment meant we might need to put both `Tuple{Type{Vector}}` and `Tuple{Type{Vector.body}}` in the map, which sigs_equal couldn't distinguish between, so it needed `simplesig` to see that one was specialized from a call to `Tuple{DataType}` while the other was a call to `Tuple{TypeConstructor}`. I think that's shouldn't be possible anymore?